modules/camera/camera_android.cpp

```diff
@@ -0,0 +1,512 @@
+	if (c_status != ACAMERA_OK) {
+		onError(this, device, c_status);
+		return false;
+	}
+
+	// Create image reader
+	const FeedFormat &feed_format = formats[selected_format];
+	media_status_t m_status = AImageReader_new(feed_format.width, feed_format.height, feed_format.pixel_format, 1, &reader);
+	if (m_status != AMEDIA_OK) {
+		onError(this, device, m_status);
+		return false;
+	}
+
+	// Get image listener
+	static AImageReader_ImageListener listener{
+		.context = this,
+		.onImageAvailable = onImage,
+	};
+	m_status = AImageReader_setImageListener(reader, &listener);
+	if (m_status != AMEDIA_OK) {
+		onError(this, device, m_status);
+		return false;
+	}
+
+	// Get image surface
+	ANativeWindow *surface;
+	m_status = AImageReader_getWindow(reader, &surface);
+	if (m_status != AMEDIA_OK) {
+		onError(this, device, m_status);
+		return false;
+	}
+
+	// Prepare session outputs
+	ACaptureSessionOutput *output = nullptr;
+	c_status = ACaptureSessionOutput_create(surface, &output);
+	if (c_status != ACAMERA_OK) {
+		onError(this, device, c_status);
+		return false;
+	}
+
+	ACaptureSessionOutputContainer *outputs = nullptr;
+	c_status = ACaptureSessionOutputContainer_create(&outputs);
+	if (c_status != ACAMERA_OK) {
+		onError(this, device, c_status);
+		return false;
+	}
+
+	c_status = ACaptureSessionOutputContainer_add(outputs, output);
+	if (c_status != ACAMERA_OK) {
+		onError(this, device, c_status);
+		return false;
+	}
+
+	// Create capture session
+	static ACameraCaptureSession_stateCallbacks sessionStateCallbacks{
+		.context = this,
+		.onClosed = onSessionClosed,
+		.onReady = onSessionReady,
+		.onActive = onSessionActive
+	};
+	c_status = ACameraDevice_createCaptureSession(device, outputs, &sessionStateCallbacks, &session);
+	if (c_status != ACAMERA_OK) {
+		onError(this, device, c_status);
+		return false;
+	}
+
+	// Create capture request
+	c_status = ACameraDevice_createCaptureRequest(device, TEMPLATE_PREVIEW, &request);
+	if (c_status != ACAMERA_OK) {
+		onError(this, device, c_status);
+		return false;
+	}
+
+	// Set capture target
+	ACameraOutputTarget *imageTarget = nullptr;
+	c_status = ACameraOutputTarget_create(surface, &imageTarget);
+	if (c_status != ACAMERA_OK) {
+		onError(this, device, c_status);
+		return false;
+	}
+
+	c_status = ACaptureRequest_addTarget(request, imageTarget);
+	if (c_status != ACAMERA_OK) {
+		onError(this, device, c_status);
+		return false;
+	}
+
+	// Start capture
+	c_status = ACameraCaptureSession_setRepeatingRequest(session, nullptr, 1, &request, nullptr);
+	if (c_status != ACAMERA_OK) {
+		onError(this, device, c_status);
+		return false;
+	}
+
+	return true;
+}
+
+bool CameraFeedAndroid::set_format(int p_index, const Dictionary &p_parameters) {
+	ERR_FAIL_COND_V_MSG(active, false, "Feed is active.");
+	ERR_FAIL_INDEX_V_MSG(p_index, formats.size(), false, "Invalid format index.");
+
+	selected_format = p_index;
```

Comment:
Should `p_parameters` be used to update the selected format?

Reply:
Linux supports setting several parameters.
https://github.com/godotengine/godot/blob/master/modules/camera/camera_feed_linux.cpp#L257-L266

For macOS, `p_parameters` is not used since `set_format` is not overridden. Similarly, Android does not use `p_parameters` as there are no custom parameters.